specification/appconfiguration/AppConfiguration/models.tsp

```diff
@@ -0,0 +1,332 @@
+@resource("snapshots")
+model Snapshot {
+  @key
+  @doc("The name of the snapshot.")
+  @visibility("read")
+  name: string;
+
+  @doc("The current status of the snapshot.")
+  @visibility("read")
+  status?: SnapshotStatus;
+
+  @doc("A list of filters used to filter the key-values included in the snapshot.")
+  @minItems(1)
+  @maxItems(3)
+  filters: KeyValueFilter[];
+
+  @doc("""
+The composition type describes how the key-values within the snapshot are
+composed. The 'key' composition type ensures there are no two key-values
+containing the same key. The 'key_label' composition type ensures there are no
+two key-values containing the same key and label.
+""")
+  @encodedName("application/json", "composition_type")
+  compositionType?: CompositionType;
+
+  @doc("The time that the snapshot was created.")
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  created?: utcDateTime;
+
+  @doc("The time that the snapshot will expire.")
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  expires?: utcDateTime;
+
+  @doc("""
+The amount of time, in seconds, that a snapshot will remain in the archived
+state before expiring. This property is only writable during the creation of a
+snapshot. If not specified, the default lifetime of key-value revisions will be
+used.
+""")
+  @maxValue(7776000)
+  @minValue(3600)
+  @encodedName("application/json", "retention_period")
+  retentionPeriod?: int32;
+
+  @doc("The size in bytes of the snapshot.")
+  @visibility("read")
+  size?: int32;
+
+  @doc("The amount of key-values in the snapshot.")
+  @visibility("read")
+  @encodedName("application/json", "items_count")
+  itemsCount?: int32;
+
+  @doc("The tags of the snapshot.")
+  tags?: Record<string>;
+
+  @doc("A value representing the current state of the snapshot.")
+  @visibility("read")
+  etag?: string;
+}
+
+@doc("""
+Enables filtering of key-values. Syntax reference:
+https://aka.ms/azconfig/docs/restapisnapshots
+""")
+model KeyValueFilter {
+  @doc("Filters key-values by their key field.")
+  key: string;
+
+  @doc("Filters key-values by their label field.")
+  label?: string;
+
+  @doc("Filters key-values by their tags field.")
+  tags?: string[];
+}
+
+@doc("Parameters used to update a snapshot.")
+model SnapshotUpdateParameters {
+  @doc("The desired status of the snapshot.")
+  status?: SnapshotStatus;
+}
+
+@doc("The result of a list request.")
+@pagedResult
+model LabelListResult {
+  @doc("The collection value.")
+  @items
+  items?: Label[];
+
+  @doc("The URI that can be used to request the next set of paged results.")
+  @nextLink
+  `@nextLink`?: string;
+}
+
+@doc("Labels are used to group key-values.")
+// FIXME: Should this be a reasource?
+// @resource("labels")
+model Label {
+  @doc("The name of the label.")
+  name?: string;
```

Comment:
I previously had this also be a resource model, but you have examples that show having null as a label `name`...so I assume this property is indeed optional? Or are the examples incorrect?

Reply:
Label can be null